src/libs/cluster-utils.js

```diff
@@ -2,53 +2,69 @@ import _ from 'lodash/fp'
 
-export const normalizeMachineConfig = ({ masterMachineType, masterDiskSize, numberOfWorkers, numberOfPreemptibleWorkers, workerMachineType, workerDiskSize }) => {
+export const normalizeRuntimeConfig = ({ cloudService, machineType, diskSize, masterMachineType, masterDiskSize, numberOfWorkers, numberOfPreemptibleWorkers, workerMachineType, workerDiskSize }) => {
   return {
-    masterMachineType: masterMachineType || 'n1-standard-4',
-    masterDiskSize: masterDiskSize || 50,
+    cloudService: cloudService || 'GCE',
+    masterMachineType: masterMachineType || machineType || 'n1-standard-4',
+    masterDiskSize: masterDiskSize || diskSize || 50,
     numberOfWorkers: numberOfWorkers || 0,
     numberOfPreemptibleWorkers: (numberOfWorkers && numberOfPreemptibleWorkers) || 0,
     workerMachineType: (numberOfWorkers && workerMachineType) || 'n1-standard-4',
     workerDiskSize: (numberOfWorkers && workerDiskSize) || 50
   }
 }
 
+export const formatRuntimeConfig = ({ sparkMode, numberOfWorkers, masterMachineType, masterDiskSize, workerMachineType, workerDiskSize, numberOfPreemptibleWorkers }) => {
+  return !!sparkMode ? {
+    cloudService: 'DATAPROC',
+    numberOfWorkers, masterMachineType,
+    masterDiskSize, workerMachineType,
+    workerDiskSize, numberOfWorkerLocalSSDs: 0,
+    numberOfPreemptibleWorkers
+  } : {
+    cloudService: 'GCE',
+    machineType: masterMachineType,
+    diskSize: masterDiskSize
+  }
+}
+
 const machineStorageCost = config => {
-  const { masterDiskSize, numberOfWorkers, workerDiskSize } = normalizeMachineConfig(config)
+  const { masterDiskSize, numberOfWorkers, workerDiskSize } = normalizeRuntimeConfig(config)
   return (masterDiskSize + numberOfWorkers * workerDiskSize) * storagePrice
 }
 
 export const findMachineType = name => {
   return _.find({ name }, machineTypes) || { name, cpu: '?', memory: '?', price: NaN, preemptiblePrice: NaN }
 }
 
-export const machineConfigCost = config => {
-  const { masterMachineType, numberOfWorkers, numberOfPreemptibleWorkers, workerMachineType } = normalizeMachineConfig(config)
-  const { price: masterPrice } = findMachineType(masterMachineType)
-  const { price: workerPrice, preemptiblePrice } = findMachineType(workerMachineType)
+export const runtimeConfigCost = config => {
+  const { cloudService, masterMachineType, numberOfWorkers, numberOfPreemptibleWorkers, workerMachineType } = normalizeRuntimeConfig(config)
+  const { price: masterPrice, cpu: masterCpu } = findMachineType(masterMachineType)
+  const { price: workerPrice, preemptiblePrice, cpu: workerCpu } = findMachineType(workerMachineType)
   return _.sum([
     masterPrice,
     (numberOfWorkers - numberOfPreemptibleWorkers) * workerPrice,
     numberOfPreemptibleWorkers * preemptiblePrice,
-    machineStorageCost(config)
+    machineStorageCost(config),
+    cloudService === 'GCE' ? 0 : masterCpu + (workerCpu * numberOfWorkers) // dataproc costs $.01 per cpu per hour
```

Comment:
Currently this is adding $1 instead of $0.01
```suggestion
    cloudService === 'GCE' ? 0 : (masterCpu + (workerCpu * numberOfWorkers)) * 0.01 // dataproc costs $.01 per cpu per hour
```